unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -370,6 +371,43 @@ export function CategoryFacetTest() {
       });
     });
 
+    describe('when calling selectValue', () => {
+      it(`when no path is selected
+      should call selectPath with the correct path`, () => {
+        test.cmp.selectValue('hello');
+        expect(test.cmp.selectPath).toHaveBeenCalledWith(['hello']);
+      });
+
+      it(`when a path is selected
+      should call selectPath with the correct path`, () => {
+        test.cmp.values.selectPath(['hello', 'there']);
+        test.cmp.selectValue('friend');
+        expect(test.cmp.selectPath).toHaveBeenCalledWith(['hello', 'there', 'friend']);
+      });
+    });
+
+    describe('when calling deselectCurrentValue', () => {
+      it(`when no value is selected
+        should warn the user`, () => {
+        test.cmp.deselectCurrentValue();
+        expect(test.cmp.logger.warn).toHaveBeenCalled();
+      });
+
+      it(`when a path with a single value is selected
+        should call clear`, () => {
+        test.cmp.values.selectPath(['hello']);
+        test.cmp.deselectCurrentValue();
+        expect(test.cmp.clear).toHaveBeenCalled();
```

Comment:
If `clear` did not belong to the class, this would be a good test. However, since `clear` is part of the class, it makes the test not very rigorous. It would be better to check the final state of the component, like the next test is doing, along with external class calls.

If we look at the logic of the `clear` method:

```
public clear() {
    if (!this.values.hasSelectedValue) {
      return;
    }

    this.logger.info('Clear facet');
    this.values.clear();
    this.changeActivePath([]);
  }
```

The key assertions are:
1. The `clear` method on this.values is called once.
2. The active path is set to an empty array.

By just checking that `clear` was called, we don't know if these important things are happening,

Reply:
It should be enough since clear is thoroughly tested in another unit test? Isn't it the purpose of a unit test, to test only what the method does, in this case it calls clear.

Reply:
Ok, as long as `clear` is tested, we can check that it's called. If we could make the check more strict in that it is called exactly one time, that would be better.

Your phrase "the purpose of a unit test" made me reflect. The key thing to keep in mind is whether the test is strong and deep enough to catch regressions.